### PR TITLE
feat: pass custom credentials as additional arguments of insights methods (v2)

### PR DESCRIPTION
**Pull request by dhayab**
See https://github.com/algolia/search-insights.js/pull/398

### CODE REVIEW

File: lib/__tests__/click.test.ts [hunk 1]
```diff
@@ -5,66 +5,109 @@ const credentials = {
   appId: "test"
 };
 
-let analyticsInstance;
```
Comment (dhayab) [inline lib/__tests__/click.test.ts:8]: A lot of noise in click / conversion / view tests, but basically it amounts to adding a test case for additional parameters which required a bit of moving stuff around to reuse instances and test payloads.

File: lib/_sendEvent.ts [hunk 2]
```diff
@@ -15,25 +16,27 @@ export function makeSendEvents(requestFn: RequestFnType) {
...
+    const events: InsightsEvent[] = eventData.map((data) => {
+      const { filters, ...rest } = data;
+
       const payload: InsightsEvent = {
         ...rest,
         userToken: data?.userToken ?? this._userToken
       };
       if (!isUndefined(filters)) {
-        payload.filters = filters.map(encodeURIComponent)
+        payload.filters = filters.map(encodeURIComponent);
```
Comment (Haroenv) [inline lib/_sendEvent.ts:27]: did prettier here fail on the before already as well?

**Review by Haroenv — COMMENTED** [review]
👍

File: lib/types.ts [hunk 3]
```diff
@@ -52,17 +52,15 @@ export type GetUserToken = MethodType<"getUserToken">;
...
 
 export type ClickedObjectIDs = MethodType<"clickedObjectIDs">;
 
 export type ClickedFilters = MethodType<"clickedFilters">;
 
-export type ConvertedObjectIDsAfterSearch = MethodType<
-  "convertedObjectIDsAfterSearch"
->;
+export type ConvertedObjectIDsAfterSearch =
+  MethodType<"convertedObjectIDsAfterSearch">;
```
[the remark this answers is not in the document]
Comment (dhayab) [reply lib/types.ts:63]: I'll reinstall the dependencies, might be a mixup due to back and forth switching with v1 / v3.

[the remark this answers is not in the document]
Comment (dhayab) [reply lib/types.ts:63]: Seems like it was not respecting prettier before, this was fixed with prettier when I edited this file.

File: lib/_sendEvent.ts (hunk 2 above)
Comment (dhayab) [reply lib/_sendEvent.ts:27]: We don't run prettier in CI, so a failure wouldn't be visible if edited in an environment that does not run prettier on file change / save.

**Review by Haroenv — APPROVED** [review]

**Review by sarahdayan — APPROVED** [review]
Yes! Can you also update the README?

**Pull request merged**